app/src/main/java/pl/pwr/zpi/bcycle/mobile/models/OngoingTrip.kt

```diff
@@ -0,0 +1,57 @@
+package pl.pwr.zpi.bcycle.mobile.models
+
+import org.threeten.bp.ZonedDateTime
+import pl.pwr.zpi.bcycle.mobile.utils.getDistance
+import pl.pwr.zpi.bcycle.mobile.utils.getTime
+
+
+data class OngoingTrip(
+    val started: ZonedDateTime,
+    var finished: ZonedDateTime?,
+    val events: MutableList<OngoingTripEvent>
+) {
+    fun getTripPoints(): List<TripPoint> {
+        return events.filter { e -> e.eventType == OngoingTripEventType.MIDPOINT }
```

Comment:
A tip for the future. You can skip braces and `return` statement in single expression functions. 
```
    fun getTripPoints(): List<TripPoint> =
        events
            .filter { e -> e.eventType == OngoingTripEventType.MIDPOINT }
            .map(OngoingTripEvent::asTripPoint)
```

Reply:
Thanks, I changed this in all the places I found, and also made more things `private`.